fedimint-server/src/net/peers.rs

```diff
@@ -237,16 +237,17 @@ where
 }
 
 pub trait PeerSlice {
-    fn peers(&self, all_peers: &BTreeSet<PeerId>) -> Vec<PeerId>;
+    fn peers(&self, all_peers: &BTreeSet<PeerId>, our_id: &PeerId) -> Vec<PeerId>;
 }
 
 impl PeerSlice for Target<PeerId> {
-    fn peers(&self, all_peers: &BTreeSet<PeerId>) -> Vec<PeerId> {
-        let set = match self {
+    fn peers(&self, all_peers: &BTreeSet<PeerId>, our_id: &PeerId) -> Vec<PeerId> {
+        let mut set = match self {
             Target::AllExcept(exclude) => all_peers.sub(exclude),
             Target::Nodes(include) => include.clone(),
         };
 
+        set.remove(our_id);
```

Comment:
> It seems to me that "peers" should not include "us" by definition, and would be more consistent to not have us in the first place. Plus we probably construct this object less often then we use it?

I'd agree that `peers` should not include ourselves. I was slightly confused by that.

> We use `peers` elsewhere to figure out what's a `threshold` number of peers. Didn't want to make yet another field in `FedimintServer`. Hopefully this code eventually gets rewritten with a better consensus algorithm anyway.

ahh, hm, ok

Reply:
*weirdly looking unsatisfied face*

Either we should rename to `all_peers` or `other_peers`, or have both. But if it's `all_peers` then we shouldn't filter "our_id" inside `fn peers()`.

It all seems not very consistent. Rewrite might come in 1 year, and in the meantime it will bite us. :)

Reply:
Yeah we are not very consistent, but in generally `peers` means all peers in the code base.

Reply:
At least we could have `fn other_peers(...)` to be more clear?